### ISSUE TITLE
Setup automated build dockerhub integration

### ISSUE THREAD

**Issue by chuyskywalker** [AHAAAAAAA/PokemonGo-Map#1366]
Now that the `develop` branch as a `Dockerfile`, it'd be useful to get an [automated build](https://docs.docker.com/docker-hub/builds/) going.

**Comment by jnovack** [comment]
If I had any say around here, (hint: I don't), the `Dockerfile` should be for production deployments.  If you wanted to do `lint`ing, or tests, the original `Dockerfile` should remain untouched.

Docker permits other `Dockerfiles` to be run with `-f`.  We can have a `linter` with tests to run before submitting a PR, and you can call it `Dockerfile.dev` as an example.

The `Dockerfile.dev` could download and build all the packages associated with `linting` and testing.  We want to keep the original `Dockerfile` as small and streamlined as possible.

**Comment by chuyskywalker** [comment]
I heartily agree with that (hence the other pull request about optimizing for file size). I actually have an uncomitted `dev/Dockerfile` which I'm using for development env which has python, node, etc, installed.

**Comment by arthurkay** [comment]
+1 for an official automated build

**Comment by JZ6** [comment]
+1 good idea

**Comment by JZ6** [comment]
@chuyskywalker are you on discord? https://discordapp.com/invite/Kzp4yky

**Comment by chuyskywalker** [comment]
I am, same name.

**Comment by medghaim** [comment]
+1

**Comment by andrewda** [comment]
+1

**Comment by jnovack** [comment]
[The Reaction icon is so we do not have to fill the comment section (and get email notifications) for every "me too".](http://tvtropes.org/pmwiki/pmwiki.php/Main/MeToo)

![image](https://cloud.githubusercontent.com/assets/387006/17105309/b2b370cc-5254-11e6-925a-67d4c3062e8a.png)

**Comment by andrewda** [comment]
@jnovack good point. Usually we do +1's on PR's so we can see how many collabs think it should be merged, but with an issue this is unnecessary.

**Comment by ultrafez** [comment]
Another way of having automated builds on Docker Hub would be to use Travis to build them ([docs here](https://docs.travis-ci.com/user/docker/#Pushing-a-Docker-Image-to-a-Registry)).
